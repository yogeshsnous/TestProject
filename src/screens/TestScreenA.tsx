import React, { useState } from "react"
import { View, TouchableOpacity, Image, Text } from "react-native"
import { getData } from "../api/Api"



type testScreenA = {
    navigation?: any,
    AppName: string,
}

const TestScreenA = (props: testScreenA) =>  {

    const onClickNaviagteTestB = async () => {

        const response = await getData("users");
        if (response.data.length > 0) {
            props.navigation.push("Share")
        }
    }

    

    return(
        <View style={{flex: 1, width: '100%', justifyContent: 'flex-start', alignItems: 'center'}}>
            <Image testID="logoImage" style={{margin: 25, width: 100, height: 100}} source={require("./circle.png")}/>

            <Text accessibilityLabel="App UI" style={{margin: 10}}> Sample App UI</Text>
            <Text testID="description" accessibilityLabel="description" style={{margin: 10}}>App Description</Text>
            <Text testID="appName" style={{margin: 10}}>{props.AppName}</Text>
            
            <TouchableOpacity
            testID="NavigateTestB"
            style={{margin: 20, height: 50, borderRadius: 25, width: 220, backgroundColor: 'green', justifyContent: 'center', alignItems: 'center'}}
            onPress={() => onClickNaviagteTestB()}
            >
                <Text style={{color: 'white'}}>Naviagte to Test B</Text>
            </TouchableOpacity>
        </View>
    )

}
export default TestScreenA;