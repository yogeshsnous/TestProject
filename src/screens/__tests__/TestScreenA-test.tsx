import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TestScreenA from '../TestScreenA';
import { render, fireEvent } from '@testing-library/react-native'; 


const mockNaviagtion = {
    navigate: jest.fn(),
    push: jest.fn(),
}


describe('Test Screen A Snapshot', () => {

    test('SnapShot tets', () => {
        const testATree = renderer.create(<TestScreenA AppName='Sample App'/>);
        expect(testATree).toMatchSnapshot();
    })

})

describe("Test screen", () => {

    const profile = {
        name: "",
        class: ''
    }

    test("Check image is present", () => {
        const {getByTestId} = render(<TestScreenA AppName=''/>)
        const image = getByTestId('logoImage');
        expect(image).toBeTruthy();
    })
    test("Check App Text is present", () => {
        const {getByLabelText} = render(<TestScreenA AppName='Rahul'/>)
        const text = getByLabelText('App UI');
        expect(text).toBeTruthy();
    })
   
    test("Check Navigation button  is present", () => {
        const {getByTestId} = render(<TestScreenA AppName='abc' />)
        const naviagteButton = getByTestId('NavigateTestB');
        expect(naviagteButton).toBeTruthy();
    })

    test("Check navigation button navigates to next screen on click", () => {
        const spyNaviagte = jest.spyOn(mockNaviagtion, 'push')
        const {getByTestId} = render(<TestScreenA AppName='XYZ' navigation={mockNaviagtion}/>)
        const naviagteButton = getByTestId('NavigateTestB');
        fireEvent.press(naviagteButton);
        expect(spyNaviagte).toHaveBeenCalled();
    })
})