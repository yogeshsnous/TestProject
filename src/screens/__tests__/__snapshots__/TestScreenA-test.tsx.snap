// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Screen A Snapshot SnapShot tets 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
      "width": "100%",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/screens/circle.png",
      }
    }
    style={
      {
        "height": 100,
        "margin": 25,
        "width": 100,
      }
    }
    testID="logoImage"
  />
  <Text
    accessibilityLabel="App UI"
    style={
      {
        "margin": 10,
      }
    }
  >
     Sample App UI
  </Text>
  <Text
    style={
      {
        "margin": 10,
      }
    }
    testID="description"
  >
    App Description
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "green",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "margin": 20,
        "opacity": 1,
        "width": 220,
      }
    }
    testID="NavigateTestB"
  >
    <Text
      style={
        {
          "color": "white",
        }
      }
    >
      Naviagte to Test B
    </Text>
  </View>
</View>
`;
